homeassistant/components/solarlog/config_flow.py

```diff
@@ -145,3 +152,63 @@ async def async_step_reconfigure(
                 }
             ),
         )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> OptionsFlow:
+        """Create the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+
+class OptionsFlowHandler(OptionsFlow):
```

Comment:
Why don't we add all inverters by default?

Reply:
Depending on the setup there may be quite some devices with no or little added value for the user. Me for example, I do have four devices: one inverter (with values of solar yield, that I already get from base setup), one gauge for energy consumption (also already integrated in base setup), one gauge for my heat pump (this one I need) and one relais (can be triggered by solar energy surplus). Based on the SolarLog manual there could be loads more devices be connected. Hence I do not think all devices should be added by default.
I thought about adding it in the configuration, but in my view the OptionFlow was a better fit (despite being more complicated to implement).

Reply:
We usually want to go with an approach where we add everything, and people can disable devices if they want

Reply:
I definitely would not like this from a user perspective, but if this is the rule...
Would it be an option to copy the step to the initial ConfigFlow, so that the user can decide in the initial setup?

Reply:
https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md

It's an old one but it explains the decision well

Reply:
Thnx. In the reference it is said this applies for data retrieved in one API call. In the case at hand, for every device at least 3 calls are needed (name, consumption, power). The library does only make the calls for enabled devices. Does this still apply?

Reply:
Perhaps this would make sense. I think the alternative to give the user the possibility to choose during the initial setup would also be user-friendly. But I would rather not like to spam the user with loads of unnecessary devices.